audb/core/load.py

```diff
@@ -558,6 +602,75 @@ def _load_tables(
     return cached_versions
 
 
+def _maybe_replace_file_extension(
+        files: typing.Union[typing.List, pd.Index],
+        format: str,
+        deps: Dependencies,
+) -> typing.Union[typing.List, pd.Index]:
+    r"""Replaces file extension if required for format.
+
+    If ``format='flac'`` and the index contains files
+    ``'f1.wav'``,
+    ``'f2.flac'``,
+    ``'f3.FLAC'``
+    the index will be updated to contain
+    ``'f1.flac'``,
+    ``'f2.flac'``,
+    ``'f3.FLAC'``
+    afterwards.
+
+    The implementation follows
+    :func:`audformat.utils.replace_file_extension`,
+    which cannot be directly used
+    as you can only provide it a file extension,
+    but not a replacement pattern.
+
+    Args:
+        files: filewise or segmented index,
+            or list of files
+        format: requested format, e.g. ``'wav'``
+        deps: dependency object
+
+    Returns:
+        index or list with replaced file formats if required
+
+    Example:
+        >>> format = 'wav'
+        >>> index = audformat.filewise_index(['f1,wav', 'f2.WAV', 'f3.flac'])
+        >>> _maybe_replace_file_extension(index, format, deps)  # doctest: +SKIP
+        Index(['f1,wav', 'f2.WAV', 'f3.wav'], dtype='string', name='file')
+
+    """  # noqa: E501
+    if format is not None and len(files) > 0:
+        # First check if we need to replace anything at all
+        extensions = set([deps.format(m) for m in deps.media])
+        extensions = [extension.lower() for extension in extensions]
+        if not (len(extensions) == 0 and extensions[0] == format):
```

Comment:
This `if` case is not really needed, but I thought it might be faster to replace index entries only if really needed.
This we can check by inspecting the format column in `deps`.